test/test_helper.rb

```diff
@@ -9,38 +9,24 @@
 
 module MiniTest
   class Test
+    def setup
+      super
+
+      path = "/tmp/#{location.tr('#', '_')}"
```

Comment:
location is a MiniTest method to get the currently tested method. It's just a way to get a unique tmp path. `Tempfile` seemed to be garbage collected too soon :(